includes/export/abstract-wc-csv-batch-exporter.php

```diff
@@ -55,11 +56,11 @@ protected function get_file_path() {
 	 */
 	public function get_file() {
 		$file = '';
-		if ( @file_exists( $this->get_file_path() ) ) {
-			$file = @file_get_contents( $this->get_file_path() );
+		if ( @file_exists( $this->get_file_path() ) ) { // phpcs:ignore Generic.PHP.NoSilencedErrors.Discouraged
+			$file = @file_get_contents( $this->get_file_path() ); // phpcs:ignore Generic.PHP.NoSilencedErrors.Discouraged, WordPress.WP.AlternativeFunctions.file_get_contents_file_get_contents, WordPress.WP.AlternativeFunctions.file_system_read_file_get_contents
 		} else {
-			@file_put_contents( $this->get_file_path(), '' );
-			@chmod( $this->get_file_path(), 0664 );
+			@file_put_contents( $this->get_file_path(), '' ); // phpcs:ignore WordPress.VIP.FileSystemWritesDisallow.file_ops_file_put_contents, Generic.PHP.NoSilencedErrors.Discouraged, WordPress.WP.AlternativeFunctions.file_system_read_file_put_contents
+			@chmod( $this->get_file_path(), 0664 ); // phpcs:ignore WordPress.VIP.FileSystemWritesDisallow.chmod_chmod, WordPress.WP.AlternativeFunctions.file_system_read_file_put_contents, Generic.PHP.NoSilencedErrors.Discouraged
```

Comment:
Instead of silencing the error and ignorig the PHPCS rule, would it be possible to check if is possible to change the file permission and call `chmod()` only if it is?